scoring/converter.py

```diff
@@ -10,12 +10,137 @@
 
 from __future__ import annotations
 
+from sr.comp.match_period import Match
 from sr.comp.scorer.converter import (
     Converter as BaseConverter,
+    InputForm,
+    OutputForm,
     parse_int,
     render_int,
+    ZoneId,
 )
+from sr.comp.types import ScoreArenaZonesData, ScoreData, ScoreTeamData, TLA
+
+from sr2025 import DISTRICTS, RawDistrict, ZONE_COLOURS
+
+
+class SR2025ScoreTeamData(ScoreTeamData):
+    left_starting_zone: bool
 
 
 class Converter(BaseConverter):
-    pass
+    """
+    Base class for converting between representations of a match's score.
+    """
+
+    def form_team_to_score(self, form: InputForm, zone_id: ZoneId) -> SR2025ScoreTeamData:
+        """
+        Prepare a team's scoring data for saving in a score dict.
+
+        This is given a zone as form data is all keyed by zone.
+        """
+        return {
+            **super().form_team_to_score(form, zone_id),
+            'left_starting_zone':
+                form.get(f'left_starting_zone_{zone_id}', None) is not None,
+        }
+
+    def form_district_to_score(self, form: InputForm, name: str) -> RawDistrict:
```

Comment:
Hrm, the returned object is not itself a whole score, however it is in the score-flavoured structure (rather than the form-flavoured structure passed in). Open to better ideas if we have any, though worth bearing in mind that this is following existing patterns for naming of these functions.

Reply:
Feel free to submit as-is,

maybe in the future we could rename "score" to "data" but I don't think it matters